client/web/compose/src/mixins/record.js

```diff
@@ -245,6 +246,23 @@ export default {
         })
     }, 500),
 
+    handleUndelete: throttle(function () {
+      this.processingUndelete = true
+      this.processing = true
+
+      return this
+        .dispatchUiEvent('beforeUndelete')
+        .then(() => this.$ComposeAPI.recordUndelete(this.record))
+        .then(() => this.dispatchUiEvent('afterUndelete'))
```

Comment:
Could you ask @tjerman if this is a registered event? If not do we need to register it

Reply:
Yes, it should be a registered event.

Reply:
Should be; It's in the list of events